README.md

```diff
@@ -34,51 +39,116 @@ We verify this package against N-2 recent versions of Go compiler. As of Septemb
+Or, if you are already using
+[Go Modules](https://github.com/golang/go/wiki/Modules), you may specify a
+version number as well:
 
 ```bash
-$ go get github.com/getsentry/sentry-go@v0.3.0
+$ go get github.com/getsentry/sentry-go@latest
 ```
 
+Check out the [list of released versions](https://pkg.go.dev/github.com/getsentry/sentry-go?tab=versions). 
+
 ## Configuration
 
-To use `sentry-go`, you’ll need to import the `sentry-go` package and initialize it with the client options that will include your DSN. If you specify the `SENTRY_DSN` environment variable, you can omit this value from options and it will be picked up automatically for you. The release and environment can also be specified in the environment variables `SENTRY_RELEASE` and `SENTRY_ENVIRONMENT` respectively.
+To use `sentry-go`, you’ll need to import the `sentry-go` package and initialize
+it with your DSN and other [options](https://godoc.org/github.com/getsentry/sentry-go#ClientOptions).
 
-More on this in [Configuration](https://docs.sentry.io/platforms/go/config/) section.
+If not specified in the SDK initialization, the
+[DSN](https://docs.sentry.io/error-reporting/configuration/?platform=go#dsn),
+[Release](https://docs.sentry.io/workflow/releases/?platform=go) and
+[Environment](https://docs.sentry.io/enriching-error-data/environments/?platform=go)
+are read from the environment variables `SENTRY_DSN`, `SENTRY_RELEASE` and
+`SENTRY_ENVIRONMENT`, respectively.
+
+More on this in the [Configuration](https://docs.sentry.io/platforms/go/config/)
+section of the official Sentry documentation.
 
 ## Usage
 
-By default, Sentry Go SDK uses asynchronous transport, which in the code example below requires an explicit awaiting for event delivery to be finished using `sentry.Flush` method. It is necessary, because otherwise the program would not wait for the async HTTP calls to return a response, and exit the process immediately when it reached the end of the `main` function. It would not be required inside a running goroutine or if you would use `HTTPSyncTransport`, which you can read about in `Transports` section.
+The SDK must be initialized with a call to `sentry.Init`. The default transport
+is asynchronous and thus most programs should call `sentry.Flush` to wait until
+buffered events are sent to Sentry right before the program terminates.
+
+Typically, `sentry.Init` is called in the beginning of `func main` and
+`sentry.Flush` is [deferred](https://golang.org/ref/spec#Defer_statements) right
+after.
+
+> Note that if the program terminates with a call to
+> [`os.Exit`](https://golang.org/pkg/os/#Exit), either directly or indirectly
+> via another function like `log.Fatal`, deferred functions are not run.
+>
+> In that case, and if it is important for you to report outstanding events
+> before terminating the program, arrange for `sentry.Flush` to be called before
+> the program terminates.
+
+Example:
 
 ```go
```

Comment:
The example now:

- Can be built with a Go compiler;
- Does something realistic;
- Has comments teasing out details (maybe something else needs to be called out?);
- Avoids the `sentry.CaptureException` followed by `sentry.Flush` anti-pattern. Typically, `sentry.Flush` shall only be called once during termination.

Reply:
@bruno-garcia could I get your :eyes: onto this example, please? Might be easier to read it in https://github.com/getsentry/sentry-go/blob/aac7497bdcff2cdb5d069ee6ae5aeb0aee49e014/README.md#usage